### ISSUE TITLE
name() call from enum constructor is wrongly assumed to be @UnderInitialization

### ISSUE THREAD

**Issue by chajath** [typetools/checker-framework#1402]
Suppose we have an enum class with a constructor:

```java
public enum MyEnum {
  TEST_ONE("abc"),
  TEST_TWO("def");
  
  private final String newName;

  MyEnum(String customData) {
    this.newName = DoSomething.withName(name());
  }
}
```

Now, nullness checker complains that the call to `name()` is not initialized:

```
error: [method.invocation.invalid] call to name() not allowed on the given receiver.
        this.newName = DoSomething.withName(name());
                                                ^
  found   : @UnderInitialization(java.lang.Enum<MyEnum>.class) @NonNull Enum
  required: @Initialized @NonNull Enum
```

This call being made inside an enum constructor, there is not much we can do to avoid this error other than marking the whole constructor to suppress warning. It will be great if the checker framework can handle cases involving enum constructors.
